File: C/chapitre6/Exercice2.c
#include <stdio.h>
#include <dirent.h>
#include <sys/stat.h>
#include <unistd.h>
#include <pwd.h>
#include <string.h>

int main(int argc, char **argv) {
    int i;
    for (i = 1; i < argc; i++) {
        DIR *d = opendir(argv[i]);
        if (d == NULL) continue;
        struct dirent *e;
        while ((e = readdir(d)) != NULL) {
            char p[1024];
            snprintf(p, sizeof(p), "%s/%s", argv[i], e->d_name);
            struct stat st;
            if (stat(p, &st) == 0 && S_ISREG(st.st_mode) && st.st_size > 1048576) {
                struct passwd *pw = getpwuid(st.st_uid);
                printf("%s %s\n", p, pw ? pw->pw_name : "?");
            }
        }
        closedir(d);
    }
    return 0;
}

File: Python/chapitre6/Exercice2.py
import os
import sys
import pwd
for d in sys.argv[1:]:
    for root, _, files in os.walk(d):
        for f in files:
            p = os.path.join(root, f)
            try:
                s = os.stat(p)
                if s.st_size > 1048576:
                    print(f"{p} {pwd.getpwuid(s.st_uid).pw_name}")
            except:
                pass